client/templates/timeout.tmpl

```diff
@@ -1,38 +1,72 @@
+{{$ClientName := (index .Vars "client")}}
+{{$clientName := (down $ClientName)}}
+{{$interfaceName := .Interface.Name}}
+{{ $decorator := (printf "%s%s" (down $clientName) .Interface.Name) }}
+{{ $Decorator := (printf "%s%s" $ClientName .Interface.Name) }}
+{{$largeTimeoutAPIs := list "adminClient.GetCrossClusterTasks" "adminClient.GetReplicationMessages"}}
+{{$longPollTimeoutAPIs := list "frontendClient.ListArchivedWorkflowExecutions" "frontendClient.PollForActivityTask" "frontendClient.PollForDecisionTask"}}
+{{$noTimeoutAPIs := list "historyClient.GetReplicationMessages" "historyClient.GetDLQReplicationMessages" "historyClient.CountDLQMessages" "historyClient.ReadDLQMessages" "historyClient.PurgeDLQMessages" "historyClient.MergeDLQMessages" "historyClient.GetCrossClusterTasks" "historyClient.GetFailoverInfo"}}
+{{/*
+ $fieldMap defines a map of the decorator struct fields
+ with field name as the key and field type as the value
+ */}}
+{{$fieldMap := dict }}
+{{ if eq $ClientName "History" }}
+    {{$fieldMap = merge $fieldMap (dict  "timeout" "time.Duration" "client" .Interface.Type) }}
+{{ else if eq $ClientName "Admin" }}
+    {{$fieldMap = merge $fieldMap (dict  "timeout" "time.Duration" "client" .Interface.Type "largeTimeout" "time.Duration") }}
+{{ else if eq $ClientName "Frontend" }}
+    {{$fieldMap = merge $fieldMap (dict "timeout" "time.Duration" "client" .Interface.Type "longPollTimeout" "time.Duration") }}
+{{ else if eq $ClientName "Matching" }}
+    {{$fieldMap = merge $fieldMap (dict "timeout" "time.Duration" "client" .Interface.Type "longPollTimeout" "time.Duration" "peerResolver" "matching.PeerResolver" "loadBalancer" "matching.LoadBalancer") }}
+{{ end }}
+
 import (
-  "context"
-  "time"
+    "context"
+    "time"
```

Comment:
it's relatively minor but: see if you can switch your IDE to use tabs instead of spaces?  fewer changes.

Reply:
Noted